### PR TITLE
feat: add sqlx for pontos testing rework error and contract identification

### LINKED ISSUE #115: feat(storage): use sqlx as the base storage for the default implementation

### LINKED ISSUE #149: Change the contract identification function
**Describe the bug**

The contract identification for ERC1155 and ERC721 is not working as expected in all cases.
It must be adjusted to be more robust.

This will involve starknet-rs rework / deep dive to try using alternatives and use the return value from the contract to know if the entry point exist or not.

### LINKED ISSUE #150: Test with katana and default storage implementation
**Describe the bug**

We must test pontos with default storage implementation with katana to ensure it runs well and detect the expected contracts.

### PR DESCRIPTION
**Pull request by glihm**
## Description

This PR focus on preparing testing locally with Katana for pontos.
The feature also comes with some refacto like:
1. Logs are now less verbose
2. Better error handling to have information of low level error at top level
3. Better contract identification for ERC721 and ERC1155
4. Add some documentation

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature (`feat:`)
- [X] 🧑‍💻 Code Refactor (`refactor:`)

## Added tests?

- [X] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [X] 📜 README.md
- [ ] 📓 Documentation
- [ ] 🙅 no documentation needed

## Closing Issues

closes #149
closes #115
closes #150

### CODE REVIEW

File: examples/migrations/0_default.sql [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+       contract_type TEXT NOT NULL,
+       event_type TEXT NOT NULL,
+       event_id TEXT NOT NULL,
+
+       PRIMARY KEY (event_id)
+);
+
+CREATE TABLE block (
+       ts BIGINT NOT NULL,
+       num BIGINT NOT NULL,
```
Comment (kwiss) [inline examples/migrations/0_default.sql:37]: what is this ?

Comment (glihm) [reply examples/migrations/0_default.sql:37]: Timestamp and number are reserved so for shorter names I used this ones. Can be changed to something else for more clarity for sure.

**Comment by kwiss** [comment]
Globally i do think we should have the migrations with the the default sqlx storage, having them split is a bit weird because we expect to have them both together

File: examples/migrations/0_default.sql (hunk 1 above)
Comment (kwiss) [reply examples/migrations/0_default.sql:37]: oh ok, yes maybe we can have something more descriptive ?

File: crates/pontos/src/managers/contract_manager.rs [hunk 2]
```diff
@@ -83,41 +83,103 @@ impl<S: Storage, C: StarknetClient> ContractManager<S, C> {
         }
     }
 
+    /// Verifies if the contracft is an ERC721, ERC1155 or an other type.
```
Comment (kwiss) [inline crates/pontos/src/managers/contract_manager.rs:86]: nitpick: typo

**Comment by kwiss** [comment]
suggestion:  also add the new contract identification in the title

**Comment by glihm** [comment]
> Globally i do think we should have the migrations with the the default sqlx storage, having them split is a bit weird because we expect to have them both together, also maybe we can have the default implementation in the exemple folder

Agreed on this concern. Having the default inside pontos is handy for testing, as we can write a pontos test without the need to use an other dependency/binary. So I would prefer everything in pontos.
Now if it makes sense to have it outside, no problem and we then have to add the example as a binary, to better identify the tests in the CI. WDYT?

**Review by kwiss — APPROVED** [review]

**Review by kwiss — APPROVED** [review]

**Pull request merged**